### PR TITLE
Add tokenizer test

### PR DESCRIPTION
**Pull request by drwpow**
You can run this test with:

```
go test -v ./internal
```

Screen Shot 2021-08-23 at 17 52 58

It should pass

### CODE REVIEW

File: internal/token_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+func TestTokenizer(t *testing.T) {
+	tests := []struct {
+		name string
+		input string
+		want []TokenType
+	}{
+		{
+		  "doctype",
+			`<!DOCTYPE html>`,
+			[] TokenType { DoctypeToken },
```
Comment (drwpow) [inline internal/token_test.go:18]: I was thinking to expand these tests, we could test longer HTML strings that yield slices of tokens. Hence the param here being a slice.

What do you think about the test setup here?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      - name: Build
+        run: GOOS=js GOARCH=wasm go build ./cmd/astro
```
Comment (drwpow) [inline .github/workflows/ci.yml:21]: This tests that the build can still happen

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+      - name: Test
+        run: GOOS=js GOARCH=wasm go test ./internal
```
Comment (drwpow) [inline .github/workflows/ci.yml:24]: This runs the unit tests (currently only in `./internal`; can update as needed)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (natemoo-re) [reply .github/workflows/ci.yml:21]: We're actually using [`tinygo`](https://tinygo.org/) to build but `GOARCH=wasm` might be easier for CI? Not sure if there's a setup action for `tinygo`

**Review by natemoo-re — APPROVED** [review]
Looks awesome, thanks for adding!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (natemoo-re) [reply .github/workflows/ci.yml:21]: Specifically `tinygo build -target=wasm -no-debug -o=astro.wasm ./cmd/astro `

**Comment by natemoo-re** [comment]
@drwpow Got all the checks passing! Going to merge. 😄

**Pull request merged**